Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,444 @@
+
+			try
+			{
+				calManager = new CalendarManager();
+				calRecord = calManager.Database.Get(Event.Uri, Convert.ToInt32(calendarEvent.ExternalID));
+				startTime = calRecord.Get<CalendarTime>(Event.Start);
+				calTime = new CalendarTime(startTime.UtcTime.AddMinutes(reminder?.TimeBefore.TotalMinutes ?? 15).Ticks);
+				calAlarm = new CalendarRecord(Alarm.Uri);
+
+				calAlarm.Set<CalendarTime>(Alarm.AlarmTime, calTime);
+				calAlarm.Set<int>(Alarm.TickUnit, (int)CalendarTypes.TickUnit.Specific);
+				calRecord.AddChildRecord(Event.Alarm, calAlarm);
+				calManager.Database.Update(calRecord);
+			}
+			finally
+			{
+				calAlarm?.Dispose();
+				calAlarm = null;
+
+				calRecord?.Dispose();
+				calRecord = null;
+
+				calManager?.Dispose();
+				calManager = null;
+			}
+			return true;
+		}
+
+		/// <summary>
+		/// Removes a calendar and all its events from the system.
+		/// </summary>
+		/// <param name="calendar">Calendar to delete</param>
+		/// <returns>True if successfully removed</returns>
+		/// <exception cref="System.ArgumentException">Calendar is read-only</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<bool> DeleteCalendarAsync(Calendar calendar)
+		{
+			throw new PlatformNotSupportedException("Tizen platform doesn't support delete calendars");
+		}
+
+		/// <summary>
+		/// Removes an event from the specified calendar.
+		/// </summary>
+		/// <param name="calendar">Calendar to remove event from</param>
+		/// <param name="calendarEvent">Event to remove</param>
+		/// <returns>True if successfully removed</returns>
+		/// <exception cref="System.ArgumentException">Calendar is read-only</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="System.InvalidOperationException">Editing recurring events is not supported</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public async Task<bool> DeleteEventAsync(Calendar calendar, CalendarEvent calendarEvent)
+		{
+			if (string.IsNullOrEmpty(calendar.ExternalID) || string.IsNullOrEmpty(calendarEvent.ExternalID))
+			{
+				return false;
+			}
+
+			var existingCal = await GetCalendarByIdAsync(calendar.ExternalID).ConfigureAwait(false);
+			if (existingCal == null)
+			{
+				return false;
+			}
+			else if (!existingCal.CanEditEvents)
+			{
+				throw new ArgumentException("Cannot delete event from readonly calendar", "calendar");
+			}
+			else
+			{
+				CalendarManager calManager = null;
+				CalendarList calList = null;
+
+				try
+				{
+					calManager = new CalendarManager();
+					calManager.Database.Delete(Event.Uri, Convert.ToInt32(calendarEvent.ExternalID));
+					calList = calManager.Database.GetAll(Event.Uri, 0, 0);
```

Comment:
Don't think there's any reason to be getting all the calendar events here? Seems like the `calList` variable and its uses can be removed entirely